controller/deployments.go

```diff
@@ -381,6 +384,67 @@ func (c *DeploymentsController) ShowSpaceEnvironments(ctx *app.ShowSpaceEnvironm
 	return ctx.OK(res)
 }
 
+// WatchEnvironmentEvents runs the watchEnvironmentEvents action.
+func (c *DeploymentsController) WatchEnvironmentEvents(ctx *app.WatchEnvironmentEventsDeploymentsContext) error {
+	c.connectEventsHandler(ctx).ServeHTTP(ctx.ResponseWriter, ctx.Request)
+	return nil
+}
+
+func (c *DeploymentsController) connectEventsHandler(ctx *app.WatchEnvironmentEventsDeploymentsContext) websocket.Handler {
+	return func(ws *websocket.Conn) {
+		defer ws.Close()
+
+		kc, err := c.GetKubeClient(ctx)
+		defer cleanup(kc)
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"err": err,
+			}, "error accessing Auth server")
+			return
```

Comment:
Is there some way we can inform the client of errors that may occur here and throughout this method?

Reply:
Ah good point. At the moment this endpoint is just repeating the results of the Kubernetes watch and listen to the client. It can also be made to send useful error messages to the client via the websocket.

As well, instead of repeating the results raw, I was thinking about whether or not we want to massage the data a little to tailor it to the UX needs, like how our other deployments endpoints are. Thoughts?

Reply:
Tailoring the responses to the frontend makes sense to me. As you said, we already do it elsewhere.